src/proxy/BaseProxy.ts

```diff
@@ -74,6 +74,19 @@ export class BaseProxy {
     }
 
     /**
+     * Encodes a request from a codec and invokes it on owner node of random partition.
+     * @param codec
+     * @param codecArguments
+     * @returns
+     */
+    protected encodeInvokeOnRandomPartition<T>(codec: any, ...codecArguments: any[]): Promise<T> {
```

Comment:
I do not think we need this method here. General approach in Hazelcast clients is to assign a random partition id to a proxy instance at first call and then use `encodeInvokeOnPartition` with that partition number for the rest of the calls in that proxy instance. You can see how it is done in hazelcast-python-client https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/proxy/replicated_map.py#L236

We try to keep implementations consistent across different clients.